### DISCUSSION TITLE
Get all Atoms in Family for Request

### DISCUSSION THREAD

**Discussion by fjur** [facebookexperimental/Recoil#1767] [Q&A]
We want to fetch all the atoms in a family and then make a POST request with these atoms to update an API.

We considered using the atom families effects to update one by one, but if a single call fails then our state and external database will be out of sync. So instead of making a call every update, we just want to send the entire family to a request to update everything, how would one do this?

Also, is there a way to "get" other atoms/selectors in the effects of an atom family? We have other atoms we need to reference to make API calls, and don't want to have to duplicate that data for each atom family item to make the call if we did individual updates from an effect. Plus when you reset an atom to its default id, having just the id isn't helpful to make an api call since if it follows openAPI spec, you don't have the other data required to make the api call.

**Comment by drarmstr** [comment]
It's a common practice to use an atom to store an "index" for another atom family.  You can access other atom/selector values from an atom effect using `getPromise()` or `getLoadable()`.

If you're dealing with batching atom syncs you may find the [`recoil-sync`](https://recoiljs.org/docs/recoil-sync/introduction) library helpful.  The NPM package isn't published yet, but the [source](https://github.com/facebookexperimental/Recoil/tree/main/packages/recoil-sync) is available for use.

**Reply by peterwiebe** [reply]
@drarmstr is it too difficult to offer an api on an Atom Family to get all atoms or atom ids? Seems a bit hackish to have to create a separate atom for keeping track of ids that needs to be kept in sync with the Family.

**Reply by 3xau1o** [reply]
would say that if this is very common case a built-in utility should be provided
